test/extended/apiserver/latency.go

```diff
@@ -0,0 +1,84 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	promhelper "github.com/openshift/origin/test/extended/util/prometheus"
+)
+
+var (
+	oc                   *exutil.CLI
+	url, bearerToken, ns string
+	execPod              *v1.Pod
+)
+
+var _ = g.Describe("[sig-instrumentation][Late] API Server latency", func() {
+	defer g.GinkgoRecover()
+
+	oc = exutil.NewCLIWithoutNamespace("apiserver-latency")
+	ns = oc.SetupNamespace()
+
+	// TODO: access Prometheus via route instead of exec pod
+	execPod = exutil.CreateExecPodOrFail(oc.AdminKubeClient(), ns, "execpod")
+	defer func() {
+		oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+	}()
+
+	g.BeforeEach(func() {
+		err := exutil.WaitForAnImageStream(
+			oc.AdminImageClient().ImageV1().ImageStreams("openshift"), "tools",
+			exutil.CheckImageStreamLatestTagPopulated, exutil.CheckImageStreamTagNotFound,
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		var ok bool
+		url, _, bearerToken, ok = promhelper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+	})
+
+	g.It("should verify that the apiserver requests latency are within expected thresholds", func() {
```

Comment:
provide a comment that indicates where you got these actual p99s. Not so I can criticize it, but so that you can collect it again.